File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int n = nums.length;
        int idx = 0;
        int [] ret = new int[n];
        
        for(int num: nums){
            if (num < pivot){
                ret[idx++] = num;
            }
        }
        for (int num: nums){
            if (num == pivot){
                ret[idx++] = num;
            }
        }
        for (int num: nums){
            if (num > pivot){
                ret[idx++] = num;
            }
        }
        return ret;
    }
}

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        ret = [0 for i in range(len(nums))]
        i = 0
        
        for num in nums:
            if num < pivot:
                ret[i] = num
                i += 1
        
        for num in nums:
            if num == pivot:
                ret[i] = num
                i += 1
                
        for num in nums:
            if num > pivot:
                ret[i] = num
                i += 1
                
        return ret
        